modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/cluster/storage/ClusterStorage.scala

```diff
@@ -66,43 +66,51 @@ object ClusterStorage {
       def getPeer(id: PeerId): F[Option[Peer]] =
         peers(id).get
 
-      def addPeer(peer: Peer): F[Unit] =
-        setPeer(peer.id)(peer.some)
+      def addPeer(peer: Peer): F[Boolean] =
+        updatePeerAndGet(peer.id) {
+          case Some(currPeer) =>
+            if (currPeer.session < peer.session) peer.some
+            else currPeer.some
+          case None => peer.some
```

Comment:
So we want `peer` in 2 cases:

- currPeer doesn't exist
- currPeer.session is older than peer.session

If you treat both of these cases as `None` (by using filter) then you can get rid of repeated `peer.some`


```scala
updatePeerAndGet {
  _.filterNot(_.session < peer.session).orElse(peer.some)
}
```